src/relay/pass/type_infer.cc

```diff
@@ -140,14 +124,40 @@ class TypeInferencer : private ExprFunctor<Type(const Expr&)> {
       return Type();
     }
   }
+
+  // this is a temporary measure to ensure all type vars are
+  // converted into fresh incomplete type vars until
+  // generalization is properly implemented
+  Type InstantiateAwayTypeVars(const Type &t) {
+    if (!t.defined()) {
+      return t;
+    }
+    auto* ft = t.as<FuncTypeNode>();
+    if (ft == nullptr) {
+      return Bind(t, instantiation_map_);
+    }
+
+    for (auto type_param : ft->type_params) {
+      instantiation_map_.Set(type_param, IncompleteTypeNode::make(TypeVarNode::Kind::kType));
+    }
+
+    Type ret_type = ft->ret_type;
+    if (!ret_type.defined()) {
```

Comment:
If I don't have this check, I have errors in several test cases, including ones related to the interpreter, but I am not sure where null return types for func types are being created. I believe it certainly shouldn't be allowed. @jroesch (or anyone else who worked on the interpreter) do you know where they could be coming from?

Reply:
Could be a bug in the visitor? I can look in depth later